xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,29 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
+    }
+    else if (g_infoManager.GetShowInfo() &&
+             CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") &&
+             g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      // If only channel info but no player controls are currently visible
+      // and setting "confirm channel switch with OK" is active, do a channel
+      // switch instead of turning on the the player controls.
+      g_application.m_pPlayer->SwitchChannel(g_application.CurrentFileItem().GetPVRChannelInfoTag());
```

Comment:
This method gets called if the user hits OK while the OSD is displayed - the event we are interested in for confirm channel switch. If the player controls are not visible then and music information OSD is visible and confirm channel switch is activate and currently displayed informattion is for a PVR supplied radio channel (confirm channel switch only works for PVR TV/radio stuff), then and only then pressing OK switches to the channel currently displayed in OSD.

Reading this myself, I suddenly get the strong feeling that there is a better place for putting this logic. ;-) Any suggestions?